keep.tex

```diff
@@ -530,38 +531,188 @@ \subsection{Concerns with active attacks}
+  generating a share of the group's private key, $s_i$. Threshold groups are
+  capped at $c$ members, and may intersect. Groups that have reached
+  this maximum size publish their public key to the blockchain. We'll
+  designate such groups as $G_{registered}$. \\
+  \textbf{Trusted setup:} & A trusted party posts a random value
+  $r_0$ to the blockchain as the beacon's first output. \\
+  \textbf{Bootstrapping:} & $mod(r_{0}, |G_{registered}|)$ is
+  used to select a registered threshold group, $G_i$, from
+  $G_{registered}$. $G_i$ signs $r_0$ and publishes the
+  result, $r_1 = threshold(r_0, s_{0\rightarrow{t}})$ where
+  $s_{0\rightarrow{t}}$ is the minimal shares necessary the the group
+  to produce a signature. \\
+  \textbf{Iteration:} & Each block published on the chain will include
+  a signature from $G_{registered}$ of the random value $r_i$. As the
+  chain grows, the signing threshold groups will change based on
+  provider availability. If any group is non-responsive up to its
+  threshold $t$, the group is removed from $G_{registered}$. \\
+  \textbf{Failure:} & Each iteration is an opportunity for a
+  group to fail to generate a valid signature. If a group $G_i$ fails
+  to sign the last iteration's random value, $G_{i+1}$ will be used
+  instead. \\
+  \hline
+\end{tabular}
+\end{table*}
 
-\subsection{The result registry}
+This work relies on the idea of threshold secret sharing schemes---
+secret sharing schemes that retain confidentiality up to some
+threshold $t$ of honest actors.
+
+Threshold signatures are a related idea. A threshold signature is a
+signature across $n$ parties that requires some minimum $t$ actively
+participating to sign. It's a similar idea to "multi-sig" as deployed
+in cryptocurrencies today.
+
+Traditional multi-sig, however, requires a smart contract on the
+blockchain to validate each signature and release funds. Threshold
+signature schemes actually require a threshold $t$ to construct a
+signature at all, removing a layer of complexity and coordination
+between parties.
+
+The use of threshold signatures means a number of participating
+signers in a signing group can be unavailable, and the signature will
+still succeed in the presence of $t$ functioning signers. This
+provides some of our beacon's required resilience in the face of
+failing or misbehaving nodes.
+
+If threshold signatures sound familiar, it might be because they're a
+core functionality keeps provide. For example, a keep signing a
+Bitcoin transaction does so using threshold ECDSA.
+
+A threshold relay is a way to chain threshold signatures to create a
+random beacon. Participants in a threshold relay form threshold
+groups. These groups generate new public keys that identify the group
+and correspond to a newly generated threshold private key, split
+across the participants.
+
+As providers join the network, they will form threshold groups. These
+groups will then sign a piece of random data from a trusted party,
```

Comment:
Who is the trusted party and how do we know that the seed is valid?

Reply:
:wave:

Reply:
We'll be bootstrapping the network. I don't want to break voice but it is worth noting in the whitepaper that the initial network contributors will be involved in the bootstrapping process. It's not *really* a trusted setup, at least not in the sense the ZCash has one, but it will matter for a few rounds